ui/src/views/Documentation/index.jsx

```diff
@@ -106,63 +182,220 @@ export default class Documentation extends Component {
     return this.findChildFromRootNode(rootNode) || rootNode;
   }
```

Comment:
I know this has already landed, but I noticed that this `findChildFromRootNode` is doing a depth-first search for the named node.  I think that's done because the "filename" structure doesn't match the TOC structure, so a tree traversal doesn't work.  Let's make those correspond so URL pathname and TOC position match one another!

Reply:
@djmitche How do you propose we do this?

Reply:
I think abandoning filenames in favor of a TOC-path would help (that TOC path is probably partly based on the filename, though).  Then the node data structure can either be a tree which is traversed following that path, or a flat Map data structure keyed by the path.

Reply:
Now that filenames are a little easier to see in the source tree, this makes a lot more sense.  I don't think the DFS itself is a problem (we don't have a lot of pages of docs).  Aside from some of the references paths, the TOC path and filename seem to be well-aligned.  And those references docs will likely change as we host those docs permanently in `ui/docs` the way @owlishDeveloper suggested today.  So I'll mark this resolved.